src/java/org/apache/cassandra/concurrent/Stage.java

```diff
@@ -124,10 +129,20 @@ public static Stage fromPoolName(String stageName)
     // Convenience functions to execute on this stage
     public void execute(Runnable command) { executor().execute(command); }
     public void execute(Runnable command, ExecutorLocals locals) { executor().execute(command, locals); }
-    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); };
+    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); }
     public <T> Future<T> submit(Callable<T> task) { return executor().submit(task); }
     public Future<?> submit(Runnable task) { return executor().submit(task); }
     public <T> Future<T> submit(Runnable task, T result) { return executor().submit(task, result); }
+    public Runnable submitIfNotShutdown(Runnable task)
```

Comment:
It feels like we could probably leave this logic in `MigrationCoordinator` so we don't duplicate the conditional and the log message?

Reply:
Well we should check before immediate submission and for the scheduled submission?

Reply:
yes, https://github.com/ekaterinadimitrova2/cassandra/pull/97/commits/bfe0283cd841a86184a8e2ac91959291b4ac4e55

Reply:
And this small change :-) https://github.com/ekaterinadimitrova2/cassandra/commit/fd9a7fda5ec573dd16374b7b42f566002a0f2843